### PR TITLE
chore: Drop useless `return;`

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
`return;` at the end of a function is quite pointless.

Also fix some C-Style Casts

---

Checklist:

* [X] Followed the [Contributing Guidelines](https://github.com/FairRootGroup/FairRoot/blob/dev/CONTRIBUTING.md)

### CODE REVIEW

File: examples/MQ/pixelAlternative/src/PixelAltFindHits.cxx [hunk 1]
```diff
@@ -183,14 +183,12 @@ void PixelAltFindHits::GetParList(TList* tempList)
...
     fDigiPar = new PixelDigiPar("PixelDigiParameters");
     tempList->Add(fDigiPar);
-
-    return;
 }
 
 void PixelAltFindHits::InitMQ(TList* tempList)
 {
     LOG(info) << "********************************************** PixelAltFindHits::InitMQ()";
-    fDigiPar = (PixelDigiPar*)tempList->FindObject("PixelDigiParameters");
```
Comment (fuhlig1) [inline examples/MQ/pixelAlternative/src/PixelAltFindHits.cxx:193]: I think in the Init there should be a dynamic_cast with a check afterwards. We get an object by name which actually can be anything.
 
This statement holds for all usage of "getting by name".

Comment (ChristianTackeGSI) [reply examples/MQ/pixelAlternative/src/PixelAltFindHits.cxx:193]: Thanks for noting this! You're right!

I added checks for all the ones I have touched.

File: examples/MQ/pixelAlternative/src/PixelAltFindHits.cxx [hunk 2]
```diff
@@ -215,10 +214,12 @@ void PixelAltFindHits::ExecMQ(TList* inputList, TList* outputList)
     //  << "," << outputList->GetName() << "), Event " << fTNofEvents; LOG(info) <<
     //  "********************************************** PixelAltFindHits::ExecMQ(), Event " << fTNofEvents; LOG(info) <<
     //  "h" << FairLogger::flush;
-    fDigis = (TClonesArray*)inputList->FindObject("PixelDigis");
+    fDigis = dynamic_cast<TClonesArray*>(inputList->FindObject("PixelDigis"));
```
Comment (fuhlig1) [inline examples/MQ/pixelAlternative/src/PixelAltFindHits.cxx:217]: For the digis I would propose to check also if the contained data class is the expected one. Later on we do a static_cast to this type.

fDigis = dynamic_cast<TClonesArray*>(inputList->FindObject("PixelDigis"));
  if (fDigis) {
    // Check if the stored data class has the right type  
    TString objectName = fDigis->GetClass()->GetName();
    if (0 != objectName.CompareTo("PixelDigi")) {
      throw std::runtime_error("no PixelDigis");
    }
else {
   throw std::runtime_error("no PixelDigi container");
}

Comment (ChristianTackeGSI) [reply examples/MQ/pixelAlternative/src/PixelAltFindHits.cxx:217]: I don't like the string based things. I tried with `InheritsFrom` (which [TCollection::AssertClass](https://root.cern/doc/v628/classTCollection.html#a54d17025bcc640ac5a4f0ffa64cc819a) uses internally).

**Pull request merged**